### PR TITLE
fix: configurable ip address for ListenAndServe, fixes #83

### LINKED ISSUE #83: Add config setting for value used for ListenAndServe
Replace the hard coded `0.0.0.0` value currently use for ListenAndServe with a new config setting added to ServiceInfo struct. 

So this change is backwards compatible,  use `Host` value when new setting value is empty. This restores previous functionality, but allows env override to explicitly set the value used for ListenAndServe as needed for different deployments.

### PR DESCRIPTION
**Pull request by charles-knox-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

The `ListenAndServe` IP address defaults to binding to an all-interface listener value of `0.0.0.0`.

Issue Number: #83 

## What is the new behavior?

This change adds a `ServerBindAddr` configurable value. (This name is up for debate, if others have input feel free)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any new imports or modules? If so, what are they used for and why?

No

## Are there any specific instructions or things that should be known prior to reviewing?

No

## Other information

N/A

### CODE REVIEW

File: bootstrap/handlers/httpserver/httpserver.go [hunk 1]
```diff
@@ -78,10 +78,11 @@ func (b *HttpServer) BootstrapHandler(
 
 	bootstrapConfig := container.ConfigurationFrom(dic.Get).GetBootstrap()
 
-	// Do not use Service.Host in the address.
-	// Using hostname it is not recommended because it will create a listener for at most one of the host's IP addresses
-	// which becomes an issue when using distributed orchestration.
-	addr := "0.0.0.0:" + strconv.Itoa(bootstrapConfig.Service.Port)
+	port := strconv.Itoa(bootstrapConfig.Service.Port)
+	addr := bootstrapConfig.Service.ServerBindAddr + ":" + port
+	if bootstrapConfig.Service.ServerBindAddr == "" {
```
Comment (lenny-intel) [inline bootstrap/handlers/httpserver/httpserver.go:83]: Add comment why defaulting to `Service.Host` when `Service.ServerBindAddr` is empty.

Comment (charles-knox-intel) [reply bootstrap/handlers/httpserver/httpserver.go:87]: Done!

**Comment by cloudxxx8** [comment]
One question here.  In each service, what should be the default value of `ServerBindAddr`, the same as `Host `, such as `edgex-core-data`? 
https://github.com/edgexfoundry/developer-scripts/blob/20145450a491363a9f20e88df44d7d9d0c97a3af/releases/nightly-build/compose-files/docker-compose-nexus-redis.yml#L394

**Comment by lenny-intel** [comment]
@cloudxxx8 , We agreed to leave the value blank, so it default to the same value as host. This way you only add an env override when you need it to be different.

**Pull request merged**